### DISCUSSION TITLE
不懂就问：这个项目下的源码有什么用，跟教程里pip安装的paddlepaddle-gpu、paddleocr什么关系？

### DISCUSSION THREAD

**Discussion by liuzh1516** [PaddlePaddle/PaddleOCR#13104] [Q&A] [answered]
paddlepaddle-gpu、paddleoc是项目源码编译出来的吗？跟用来做ocr、版式识别的模型又是什么关系呢？新人请大神解惑

**Comment by GreatV** [comment, marked as answer]
PaddlePaddle是由百度开发的深度学习平台，而PaddleOCR是基于PaddlePaddle的OCR工具包。这里对你的问题进行详细解答：

1. **PaddlePaddle和PaddleOCR的关系**：
    - **PaddlePaddle**（PArallel Distributed Deep LEarning）是一个开源的深度学习框架，支持GPU和CPU训练。
    - **PaddleOCR**是基于PaddlePaddle的OCR（光学字符识别）系统，它提供了一系列预训练模型和工具，用于文本检测和识别任务。

2. **paddlepaddle-gpu和paddleocr是项目源码编译出来的吗**？
    - **paddlepaddle-gpu**：是PaddlePaddle框架的GPU版本，可以直接通过pip安装，例如`pip install paddlepaddle-gpu`。你也可以选择从源码编译，如果需要特定的优化或定制。
    - **PaddleOCR**：同样可以直接通过pip安装，例如`pip install paddleocr`。你也可以从PaddleOCR的GitHub仓库下载源码，并进行编译和安装。

3. **用来做OCR、版式识别的模型是什么关系**？
    - PaddleOCR提供了多种预训练模型，用于不同的OCR任务，比如文本检测、文本识别、版式分析等。这些模型基于PaddlePaddle框架进行训练和推理。
    - OCR（Optical Character Recognition）模型主要用于检测和识别图像中的文本内容。
    - 版式识别模型可以用来分析文档的布局，比如段落、标题、表格等。这些模型通常包括文本检测模型、文本识别模型和版式分析模型，它们协同工作来完成复杂的文档理解任务。

总结来说，PaddlePaddle是深度学习框架，PaddleOCR是其上的一个OCR工具包，它们提供了预训练模型和工具，方便你进行OCR和版式识别任务。你可以选择直接安装或从源码编译，具体取决于你的需求和场景。

**Reply by liuzh1516** [reply]
感谢解答，很清晰！
